llvm-project/stage1.cmake

```diff
@@ -1,45 +1,134 @@
+# Perhaps missing alpine in the triple causes some incompatibility?
 set(LLVM_DEFAULT_TARGET_TRIPLE "x86_64-alpine-linux-musl" CACHE STRING "")
+
+# Use Alpine's lld as the stage0 linker to link everything.
 set(LLVM_ENABLE_LLD ON CACHE BOOL "")
+
+# For stage 1, just build clang and LLD, which are used by the "runtimes build"
+# to build the runtime libraries of LLVM.
 set(LLVM_ENABLE_PROJECTS "clang;lld;" CACHE STRING "")
+
+# LLVM "runtimes" build.
 set(LLVM_ENABLE_RUNTIMES "compiler-rt;libcxx;libcxxabi;libunwind;" CACHE STRING "")
+
+# FORCE_ON causes the build to fail if zlib is not found in the environment
+# during configuration, rather than much later during link.
 set(LLVM_ENABLE_ZLIB "FORCE_ON" CACHE STRING "")
+
+# Just build stage1 to target the host. It's not the end product, so it won't
+# be able to target all of the kernel targets we can build.
 set(LLVM_TARGETS_TO_BUILD "host;" CACHE STRING "")
+
+# Stage1 clang should default to lld rather than BFD.
+# TODO: it seems this was ignored by the runtimes build...
+# otherwise why would CMAKE_EXE_LINKER_FLAGS be necessary?
 set(CLANG_DEFAULT_LINKER "lld" CACHE STRING "")
+# Stage1 clang should default to compiler-rt rather than libgcc.
 # TODO: it seems this was ignored by the runtimes build...
 set(CLANG_DEFAULT_RTLIB "compiler-rt" CACHE STRING "")
+
+# Disable arc migrate. We don't use that, ever.
 set(CLANG_ENABLE_ARCMT OFF CACHE BOOL "")
+
+# Disable static analyzer. Don't need it for stage1.
 set(CLANG_ENABLE_STATIC_ANALYZER OFF CACHE BOOL "")
+
+# Disable plugin support. Don't need it, ever.
 set(CLANG_PLUGIN_SUPPORT OFF CACHE BOOL "")
+
+# The compiler builtins are necessary.
 set(COMPILER_RT_BUILD_BUILTINS ON CACHE BOOL "")
+
+# GWP ASAN fails to build without libexecinfo-dev. Don't need it for stage1.
 set(COMPILER_RT_BUILD_GWP_ASAN OFF CACHE BOOL "")
+
+# Don't need libfuzzer, ever.
 set(COMPILER_RT_BUILD_LIBFUZZER OFF CACHE BOOL "")
+
+# Don't need memprof, ever.
 set(COMPILER_RT_BUILD_MEMPROF OFF CACHE BOOL "")
+
+# Don't need ORC, ever.
 set(COMPILER_RT_BUILD_ORC OFF CACHE BOOL "")
+
+# Disable profiling support. Not necessary for stage1.
 set(COMPILER_RT_BUILD_PROFILE OFF CACHE BOOL "")
+
+# Disable sanitizer support. Not necessary for stage1.
 set(COMPILER_RT_BUILD_SANITIZERS OFF CACHE BOOL "")
+
+# Don't need xray.
 set(COMPILER_RT_BUILD_XRAY OFF CACHE BOOL "")
+
+# Compiler-rt is meant as a replacement to libgcc.
 set(COMPILER_RT_HAS_GCC_S_LIB OFF CACHE BOOL "")
+
+# Who needs docs anyways? Can't RTFM if there is no FM.
 set(LIBUNWIND_INCLUDE_DOCS OFF CACHE BOOL "")
+
+# We don't plan to run the tests, disable them.
 set(LIBUNWIND_INCLUDE_TESTS OFF CACHE BOOL "")
+
+# We need the headers available when we install-libunwind.
 set(LIBUNWIND_INSTALL_HEADERS ON CACHE BOOL "")
+
+# Libunwind should use compiler-rt rather than libgcc.
 set(LIBUNWIND_USE_COMPILER_RT ON CACHE BOOL "")
+
+# TODO: I forget why I enabled this.
```

Comment:
We need to enable this to break the cycle between libc, libc++, and the unwinder.